dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromPrompt.cs

```diff
@@ -434,6 +430,19 @@ private void CaptureUsageDetails(string? modelId, IReadOnlyDictionary<string, ob
             s_invocationTokenUsagePrompt.Record(promptTokens, in tags);
             s_invocationTokenUsageCompletion.Record(completionTokens, in tags);
         }
+        else if (jsonObject.TryGetProperty("InputTokenCount", out var inputTokensJson) &&
```

Comment:
Maybe not related to this PR, but I'm wondering why this method exists in `KernelFunctionFromPrompt` rather than specific connector? Here we were handling cases where model returns `PromptTokens`-`CompletionTokens`, now we are adding `InputTokenCount`-`OutputTokenCount`, and the question is - how many more options we will have to support in the future here. If AI model returns usage information under different name - we won't capture it. 

Also, usage details won't be recorded/logged if Azure OpenAI service is used directly (but there is a chance this information is already captured on Azure SDK side).

Reply:
We do have in the connectors different meters that do calculate the token counts including the Total. This method is specific for token usage for function invocation and the name of the meter is also different. Given that, I also identified and removed from the `KernelFunctionFromPrompt` the Log information about the tokens that are already performed by the connectors and was unnecessary.

>  If AI model returns usage information under different name - we won't capture it.
We plan to bring the `Usage` as an abstraction so there's no need to have such checks as this will be resolved for all connectors.

PS: I was unaware of this check before and will be careful on setting the metadata correctly for other connectors Usage.